bodhi/tests/server/test_notifications.py

```diff
@@ -81,3 +83,112 @@ def test_with_cert_prefix(self, init, info):
+        notifications.publish('demo.topic', {'such': 'important'})
+        session = Session()
+        self.assertIn('fedmsg', session.info)
+        self.assertEqual(session.info['fedmsg']['demo.topic'], [{'such': 'important'}])
+        mock_init.assert_called_once_with()
+
+    @mock.patch.dict('bodhi.server.config.config', {'fedmsg_enabled': True})
+    @mock.patch('bodhi.server.notifications.fedmsg.publish')
+    def test_publish_force(self, mock_fedmsg_publish, mock_init):
+        """Assert publish with the force flag sends the message immediately."""
+        notifications.publish('demo.topic', {'such': 'important'}, force=True)
+        session = Session()
+        self.assertEqual(dict(), session.info)
+        mock_fedmsg_publish.assert_called_once_with(
+            topic='demo.topic', msg={'such': 'important'})
+        mock_init.assert_called_once_with()
+
+
+@mock.patch.dict('bodhi.server.config.config', {'fedmsg_enabled': True})
+@mock.patch('bodhi.server.notifications.init', mock.Mock())
+@mock.patch('bodhi.server.notifications.fedmsg.publish')
+class TestSendFedmsgsAfterCommit(base.BaseTestCase):
+
+    def test_no_fedmsgs(self, mock_fedmsg_publish):
+        """Assert nothing happens if messages are not explicitly published."""
+        session = Session()
+        session.add(models.Package(name=u'ejabberd'))
+        session.commit()
+
+        self.assertEqual(0, mock_fedmsg_publish.call_count)
+
+    def test_commit_aborted(self, mock_fedmsg_publish):
+        """Assert that when commits are aborted, messages aren't sent."""
+        session = Session()
+        session.add(models.Package(name=u'ejabberd'))
+        session.commit()
+
+        session.add(models.Package(name=u'ejabberd'))
+        notifications.publish('demo.topic', {'new': 'package'})
+        self.assertRaises(exc.IntegrityError, session.commit)
+        self.assertEqual(0, mock_fedmsg_publish.call_count)
+
+    def test_single_topic_one_message(self, mock_fedmsg_publish):
+        """Assert a single message for a single topic is published."""
+        session = Session()
+        session.add(models.Package(name=u'ejabberd'))
+        notifications.publish('demo.topic', {'new': 'package'})
+        session.commit()
+        mock_fedmsg_publish.assert_called_once_with(
+            topic='demo.topic', msg={'new': 'package'})
+
+    def test_empty_commit(self, mock_fedmsg_publish):
+        """Assert a single message for a single topic is published."""
+        # Ensure nothing at all is in our session
+        Session.remove()
+        session = Session()
+        notifications.publish('demo.topic', {'new': 'package'})
+        session.commit()
+        mock_fedmsg_publish.assert_called_once_with(
+            topic='demo.topic', msg={'new': 'package'})
+
+    def test_repeated_commit(self, mock_fedmsg_publish):
+        """Assert a single message for a single topic is published."""
```

Comment:
Many of these tests have this same docblock - can you make sure the tests all have accurate docblocks?